### PR TITLE
Update Kotlin to 2.0.0-dev-4562 (tests fail)

### PR DESCRIPTION
**Pull request by JavierSegoviaCordoba**
Any hint about how to fix this?

Looks like IR extensions are not being executed.

#### Last version that is working (tests fail but IR extensions are executed):
- 1.9.20-dev-5788

#### List of versions that are not working:
- 1.9.20-Beta2
- 1.9.20-dev-6571
- 1.9.30-dev-2548
- 2.0.0-dev-4562

### CODE REVIEW

File: test/org/demiurg906/kotlin/plugin/runners/BaseTestRunner.kt [hunk 1]
```diff
@@ -28,6 +33,13 @@ abstract class BaseTestRunner : AbstractKotlinCompilerTest() {
 
 fun TestConfigurationBuilder.commonFirWithPluginFrontendConfiguration() {
     baseFirDiagnosticTestConfiguration()
+    commonConfigurationForTest(
```
Comment (demiurg906) [inline test/org/demiurg906/kotlin/plugin/runners/BaseTestRunner.kt:36]: 1. `baseFirDiagnosticTestConfiguration` partially duplicates `commonConfigurationForTest`
2. `commonConfigurationForTest` should not be called for diagnostic tests

**Review by demiurg906 — CHANGES_REQUESTED** [review]

File: test/org/demiurg906/kotlin/plugin/runners/BaseTestRunner.kt (hunk 1 above)
Comment (JavierSegoviaCordoba) [reply test/org/demiurg906/kotlin/plugin/runners/BaseTestRunner.kt]: For 1, `baseFirDiagnosticTestConfiguration` is used on diagnostic too, so the duplicated logic can't be deleted based on 2.

For 2, I have moved it to the configuration of `AbstractBoxTest`.

**Comment by JavierSegoviaCordoba** [comment]
The issue about tests failing due `MISSING_DEPENDENCY_SUPERCLASS` can be related to adding the supertypes with [`superTypeRefs.add(session.builtinTypes.anyType)`](https://github.com/demiurg906/kotlin-compiler-plugin-template/blob/93b5fcb8acac35d7748d04851d4899392fc1317d/src/org/demiurg906/kotlin/plugin/fir/SimpleClassGenerator.kt#L55C13-L55C60) instead of using the `FirSupertypeGenerationExtension` extension?

**Comment by JavierSegoviaCordoba** [comment]
The issue is hidden if I add `// !DIAGNOSTICS: -UNUSED_PARAMETER -UNUSED_VARIABLE -MISSING_DEPENDENCY_SUPERCLASS` as it is done in some tests in the Kotlin repo. Probably a bad solution.

I have tried to create a `FirSupertypeGenerationExtension` with no luck (the predicate is registered but the extension functions are not called).

**Comment by JavierSegoviaCordoba** [comment]
Looks like it is not necessary to add the `Any` supertype anymore with the new APIs

```kotlin
/**
 * Adds [type] as supertype for constructed class
 *
 * If no supertypes are declared [kotlin.Any] supertype will be
 *   added automatically
 */
public fun superType(type: ConeKotlinType) {
    superTypeProviders += { type }
}
```

**Comment by demiurg906** [comment]
I've updated version and API examples manually

**Pull request closed without merging**